src/graphics/pipeline.rs

```diff
@@ -1,49 +1,34 @@
-use std::marker::PhantomData;
-
-use wgpu::{
-    BindGroupLayout, ColorTargetState, DepthStencilState, Device, PrimitiveState, RenderPipeline,
-};
+use wgpu::{BindGroupLayout, ColorTargetState, Device, RenderPipeline};
 
 use super::{
-    resources::{vertex::Vertex, Resources},
+    resources::{vertex::Vertex, Resources, COLOR_FORMAT, DEPTH_FORMAT},
     Client,
 };
 
-mod resampling;
+use state::PipelineState;
+
+mod state;
 
 pub struct Pipelines {
     pub resampling: RenderPipeline,
-    // pub lines: RenderPipeline,
+    pub streamline: Option<RenderPipeline>,
     // pub post_processing: RenderPipeline,
 }
 
 impl Pipelines {
     pub fn new(res: &Resources, client: &Client) -> Self {
-        let fmt = res.target_texture.format;
-
+        let streamline = if res.fibers.is_some() {
+            Some(create_pipeline(state::streamline(), res, client))
+        } else {
+            None
+        };
```

Comment:
I usualy write

    let streamline = res.fibers.map(|_| create_pipeline(state::streamline(), res, client));

but this will consume the option, so maybe

    let streamline = res.fibers.as_ref().map(|_| ...);
    OR
    let streamline = res.fibers.is_some().then_some(|| ...)

but none is perfect ;) Do as you wish.

Reply:
I tried this at first but with `rustfmt` running it takes almost the same space.
```rust
        let streamline = res
            .fibers
            .as_ref()
            .map(|_| create_pipeline(state::streamline(), res, client));

```
After reconsidering I think this way is prettier, I'll update this.